### PR TITLE
Refactor/LinkDetectionModule

### PR DESCRIPTION
**Pull request by marianorodriguez**
LinkDetectionModule refactor:
- moved `sanitizeXML` function to utils file.
- removed `xmldom` unused dependency.
- removed `parseXmlToObject` unused function.
- used `xml-stream` to process dumppdf output and find links inside it.
- modified 'textual links' RegExp to also match URLs starting with `www.`

### CODE REVIEW

File: server/src/utils/DumpPdfUtils.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  });
+}
+
+/*
+  this looks that a particular nested structure defined in @param keys exists on @param o.
+  ex:
+    hasNestedKeys({ foo: { bar: { baz: 10 } } }, ['foo', 'bar', 'baz']) === true, structure "o.foo.bar.baz" is valid
+    hasNestedKeys({ foo: { bar: { baz: 10 } } }, ['foo', 'baz']) === false, structure "o.foo.baz" is not valid
+*/
+function hasNestedKeys(o: any, keys: string[]): boolean {
```
Comment (jvalls-axa) [inline server/src/utils/DumpPdfUtils.ts]: `o` should be node or element or whatever that makes easy to understand function signature ??

File: server/src/utils/DumpPdfUtils.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+}
+
+function xmlNodeToLink(node: any): DumpPdfLink {
+  const rect = node.dict.value[node.dict.key.findIndex(k => k === 'Rect')];
+  if (!rect || !hasNestedKeys(rect, ['list', 'number'])) {
+    return null;
+  }
+  const [x1, y1, x2, y2] = node.dict.value[node.dict.key.findIndex(k => k === 'Rect')].list.number
+    .map(x => parseInt(x, 10));
+  const A = node.dict.value[node.dict.key.findIndex(k => k === 'A')];
```
Comment (jvalls-axa) [inline server/src/utils/DumpPdfUtils.ts]: same comment....

Comment (marianorodriguez) [reply server/src/utils/DumpPdfUtils.ts]: solved in https://github.com/axa-group/Parsr/pull/376/commits/ab0f5da1437acaa78cd562112c2f6ca207196e7e

File: server/src/utils/DumpPdfUtils.ts (hunk 1 above)
Comment (marianorodriguez) [reply server/src/utils/DumpPdfUtils.ts]: solved in https://github.com/axa-group/Parsr/pull/376/commits/ab0f5da1437acaa78cd562112c2f6ca207196e7e

**Pull request merged**